hudi-common/src/main/java/org/apache/hudi/common/table/view/AbstractTableFileSystemView.java

```diff
@@ -259,6 +259,8 @@ private void resetFileGroupsReplaced(HoodieTimeline timeline) {
   public void close() {
     try {
       writeLock.lock();
+      this.metaClient = null;
+      this.visibleCommitsAndCompactionTimeline = null;
```

Comment:
Release the resource eagerly.

Reply:
Does the `writeLock` set to null?

Reply:
No, it is light-weight and would be garbge collected when the whole fs view is not referenced by any other objects.